Update: changelog for 15.0-beta2 (#13738)
OpenTTD/OpenTTD: changelog.md

- Change: [MacOS] Put the icon in a rounded rectangle (#13446)
- Change: [Script] GetWaypointID to return the StationID of any waypoint (#13407)
- Change: Draw company manager face jacket after collar (#13390)
- Change: Don't distinguish between bus and truck stops when removing them (#13384)
- Change: [Script] Rename BridgeID to BridgeType in the script API (#13352)
- Change: Add fonts document to help window (#13305)
- Change: Log changes to sandbox settings (#13267)
- Change: When player joins network company, use its name instead of number in chat (#13263)
- Change: [Win32] Draw window title bar according to current Windows light/dark theme (#13196)
- Change: Restore wider spacers in main toolbars (#12039)
- Fix: NewGRF Global variables 0D, 0E and 1E refer to wrong GRFFile (#13986)
- Fix #13980: Allow diagonal selection for road convert (#13983)
- Fix: Validate raw strings from game-scripts, and strip invalid and control characters (#13976)
- Fix: Capitalise "Disabled" for the "maximum non-sticky open windows" setting (#13975)
- Fix: Fixed accidental changes to autorail/road AI behavior (#13968)
- Fix: Frame widget with label had incorrect spacing (#13967)

> - Fix: StringFilter included quotes in the search and failed (#13965)

Reviewer: This PR number talks about something else: Replace Utf8Encode and other C-style string construction with bounds checking StringBuilder class

Reviewer 2: It's one of the commits from that PR, should have split better.

Reviewer 3: The commit is a part of that PR. Not sure if this is actually just a fix inside the PR, so I could drop it.

- Change: Log changes to sandbox settings (#13267)
- Change: When player joins network company, use its name instead of number in chat (#13263)
- Change: Disable service interval widgets for non-owned vehicles (#13260)
- Change: [NewGRF] Use deferred string mapping for bridge and currency names (#13230)
- Change: Restore wider spacers in main toolbars (#12039)
- Fix #13725: Use proper query strings for changing timetable values (#13737)
- Fix #11226: Don't draw story page elements that won't be visible (#13736)
- Fix: More AI than max_no_competitors could start with competitors_interval=0 (#13670)
- Fix: League table window ignored the minimal size in its widget description (#13629)
- Fix: Incorrect snow density when making rocks snowy (#13626)
- Fix #12925: Prevent cost estimates for settings changes (#13550)
- Fix: [Script] Report errors happening during 'Load()' (#13537)
- Fix: [Script] Improve type checking of parameters (#13522)
- Fix: [Script] Don't set CommandCallback for asynchronous commands (#13501)
- Fix: Missing error messages with sell- and autoreplace-all commands (#13469)
- Fix: Too many trees when generating trees at same height (#13460)

> - Fix #12912: CompanyProperties::inaugurated_year_calendar not saved (#13448)

Reviewer 4: ```suggestion
- Fix #12912: Company inaugurated year in wallclock mode was not saved (#13448)
```

- Fix: [Script] Wrong return value for failed preconditions Vehicle::CloneVehicle (#13445)
- Fix #13140: Scale initial industry production estimate by cargo scale (#13427)
- Fix #13384: Crash when remove bus/truck stop tool used on road waypoints (#13391)
- Fix #12987: Historical houses now always spawn completed (#13332)
- Fix: [Win32] Font detection didn't work for locales not supporting code pages (#13306)

> - Fix #13293: Incorrect GRFStringID used for industry cargo suffix text (#13294)

Reviewer 4: ```suggestion
```

post-beta1

## 15.x

### 15.0-beta2 (2025-03-04)

- Feature: Place rocks on "too steep" tiles when fixing slopes (#13462)
- Feature: Prevent towns from upgrading individually-placed houses (#13270)

> - Feature: [Win32] Draw window title bar according to current Windows light/dark theme (#13196)

Reviewer 4: Change/Add/Fix. Probably not a feature

> - Feature: Support converting 32bpp-only sprites to indexed 8bpp (#11602)

Reviewer 4: Probably too technical for changelog purposes. Or perhaps could be rephrased in a more user-friendly way?

Author: `- Feature: Convert 32bpp-only sprites to 8bpp when needed (#11602)`? I'm not sure how  to phrase it more user-friendly without it becoming a long story

- Add: Show cargo types produced by building in house picker (#13561)
- Add: [Script] Saving/loading ScriptList (#13556)
- Add: Press Ctrl to build diagonal canals in game mode (#13432)
- Add: Sandbox settings to Sandbox Options window (#13268)
- Add: Setting to allow placing houses manually in-game (#13266)
- Add: [Script] Event for when a company's president name changes (#13208)
- Add: Ability to toggle visibility of station signs by facility (#13207)

> - Add: 2D map scrolling on win32 (#13172)

Reviewer 4: ```suggestion
- Feature: [Win32] 2D map scrolling (#13172)
```

(& move)

> - Add: NewGRF Badges (#13073)

Reviewer 4: Feature

- Add: [Script] ScriptEventCompanyRename (#12878)
- Add: Ukrainian Hryvnia currency (#12877)

> - Change: Show the type of land covered by snow (#13659)

Reviewer 4: ```suggestion
```

covered by snowy rocks imo

Author: That's not in yet, AFAICT.

Reviewer 4: That's not my fault :p

- Change: Make tree placement at world generation look more organic (#13515)
- Change: Have a better title for the Online Content window (#13505)
- Change: [MacOS] Put the icon in a rounded rectangle (#13446)
- Change: [Script] GetWaypointID to return the StationID of any waypoint (#13407)
- Change: Draw company manager face jacket after collar (#13390)
- Change: Don't distinguish between bus and truck stops when removing them (#13384)
- Change: [Script] Rename BridgeID to BridgeType in the script API (#13352)
- Change: Add fonts document to help window (#13305)

> - Change: Remove support for SDL1.2 (#13298)

Reviewer 4: ```suggestion
- Remove: Support for SDL1.2 (#13298)
```

& move

- Change: Log changes to sandbox settings (#13267)
- Change: When player joins network company, use its name instead of number in chat (#13263)
- Change: Disable service interval widgets for non-owned vehicles (#13260)

> - Change: [NewGRF] Use deferred string mapping for bridge and currency names (#13230)

Reviewer 4: ```suggestion
```

seems like a codechange to me

- Add: Show hyperlink destination tooltips in text window (#13742)
- Add: [Script] Saving/loading ScriptList (#13556)
- Add: Press Ctrl to build diagonal canals in game mode (#13432)
- Add: Sandbox settings to Sandbox Options window (#13268)
- Add: Setting to allow placing houses manually in-game (#13266)
- Add: [Script] Event for when a company's president name changes (#13208)
- Add: Ability to toggle visibility of station signs by facility (#13207)
- Add: [Script] ScriptEventCompanyRename (#12878)
- Add: Ukrainian Hryvnia currency (#12877)
- Change: Make tree placement at world generation look more organic (#13515)
- Change: [MacOS] Put the icon in a rounded rectangle (#13446)
- Change: [Script] GetWaypointID to return the StationID of any waypoint (#13407)
- Change: Draw company manager face jacket after collar (#13390)
- Change: Don't distinguish between bus and truck stops when removing them (#13384)
- Change: [Script] Rename BridgeID to BridgeType in the script API (#13352)
- Change: Add fonts document to help window (#13305)
- Change: Log changes to sandbox settings (#13267)
- Change: When player joins network company, use its name instead of number in chat (#13263)

> - Change: Disable service interval widgets for non-owned vehicles (#13260)

Reviewer 2: ```suggestion
- Change: Disable service interval widgets for non-owned vehicles (#13260)
+ Fix: Disable service interval widgets for non-owned vehicles (#13260)
```

Author: I reckon you intended to change the `Change` to `Fix`. I've also moved it to the `Fix` section.

## 15.x

> ### 15.0-beta2 (2025-03-15)

Reviewer 5: ```suggestion
### 15.0-beta2 (2025-03-29)
```

- Feature: Snow-covered rocks are now visible (#13627)

> - Feature: Place rocks on "too steep" tiles when fixing slopes (#13462)

Reviewer 5: ```suggestion
- Feature: Generate more rocks on steep slopes during map generation or heightmap import (#13462)
```

- Feature: Prevent towns from upgrading individually-placed houses (#13270)

> - Feature: [Win32] 2D map scrolling (#13172)

Reviewer 5: ```suggestion
- Feature: [Win32] Touchpad two-finger map scrolling (#13172)
```

- Feature: NewGRF Badges (#13073)

> - Feature: Convert 32bpp-only sprites to 8bpp when needed (#11602)

Reviewer 5: I'm not sure this is a `Feature`, because who chooses to use the 8bpp blitter? (Suggestion will need to be moved to the proper order if accepted)

```suggestion
- Add: Convert 32bpp-only sprites to 8bpp when needed (#11602)
```

Reviewer: This one is relevant to NewGRF authors, not players. But we don't have a separate changelog for that.

- Fix: [NewGRF] Plurals and genders did not work in strings with cases or substrings (#13853, #13852)
- Fix: [NewGRF] String parameter stack and case selection were not processed for control code 0x81 (#13851)
- Fix #13839: Incorrect colour of first company legend in smallmap window (#13841)
- Fix: i circumflex width in TrueType small font (#13836)
- Fix: Don't show owner of non-existent road (#13824)
- Fix: Error message window timeout doesn't match setting (#13812)
- Fix #13795: Crash in vehicle list of 32-bit platforms (#13796)
- Fix: [Script] Company rename event sometimes had the wrong name (#13794)
- Fix: Improve manager face randomisation (#13776)
- Fix #13740: [Script] Handle implicit orders for jump orders (#13753)
- Fix #13749: Default service intervals were not updated when changing timekeeping unit (#13751)
- Fix #13725: Use proper query strings for changing timetable values (#13737)
- Fix #11226: Don't draw story page elements that won't be visible (#13736)
- Fix: More AI than max_no_competitors could start with competitors_interval=0 (#13670)
- Fix: League table window ignored the minimal size in its widget description (#13629)

> - Fix: Incorrect snow density when making rocks snowy (#13626)

Reviewer 5: Not player-facing until snowy rocks were added, which was after beta1.

```suggestion
```

Author: How I read the PR is that it happened before 'snowy' rocks. If a rocky tile were made snowy, it would become fully snowed and then it would thaw a bit. In effect causing a white 'flash' for the rocky tiles.

Reviewer: Yes, it was visible before snowy rocks, only with a NewGRF that introduces varying snow line.

- Fix #12925: Prevent cost estimates for settings changes (#13550)
- Fix: [Script] Report errors happening during 'Load()' (#13537)
- Fix: [Script] Improve type checking of parameters (#13522)
- Fix: [Script] Don't set CommandCallback for asynchronous commands (#13501)
- Fix: Missing error messages with sell- and autoreplace-all commands (#13469)
- Fix: Too many trees when generating trees at same height (#13460)
- Fix #12912: Company inaugurated year in wallclock mode was not saved (#13448)
- Fix: [Script] Wrong return value for failed preconditions Vehicle::CloneVehicle (#13445)
- Fix #13140: Scale initial industry production estimate by cargo scale (#13427)
- Fix #13384: Crash when remove bus/truck stop tool used on road waypoints (#13391)
- Fix #12987: Historical houses now always spawn completed (#13332)
- Fix: [Win32] Font detection didn't work for locales not supporting code pages (#13306)
- Fix: Restore ability to disable service interval (#13281)
- Fix: Hide company settings from console commands (#13269)
- Fix: Disable service interval widgets for non-owned vehicles (#13260)
- Fix #13225: Cargo payment graph key toggled wrong data sets (#13226)

> - Fix: Rail station tile flags were not set early enough (#13203)

Reviewer 5: Not really player-facing, unless I'm reading the PR wrong.

```suggestion
```

Author: It's fixing a crash triggered by an user loading an old save game.

Reviewer: It fixes a potential crash when loading saves, but that would only have occurred for a short time in the master branch, not since 14.x

Author: It's after 15.0-beta1 though, and I'm not seeing anything that would've introduced this issue after 15.0-beta1. So it's a fix of an issue in release, yes it's a beta but so are many of the other bug fixes in here.

Reviewer: Fair, Nov 15th to Dec 27th is actually longer than I remembered, and so yes that affects 15.0-beta1.

- Add: [Script] ScriptEventCompanyRename (#12878)
- Add: Ukrainian Hryvnia currency (#12877)
- Add: Convert 32bpp-only sprites to 8bpp when needed (#11602)

> - Change: [Script] Start GS (but don't run it) when generating world in scenario editor (#13961)

Reviewer 2: Isn't this a "Fix"?

Reviewer 3: Good question. The PR had it as Change, probably because it doesn't mention any real problem that is fixed, just an inconsistency. I'll wouldn't mind moving it to Fix though.

- Change: When player joins network company, use its name instead of number in chat (#13263)
- Change: [Win32] Draw window title bar according to current Windows light/dark theme (#13196)
- Change: Restore wider spacers in main toolbars (#12039)
- Fix: NewGRF Global variables 0D, 0E and 1E refer to wrong GRFFile (#13986)
- Fix #13980: Allow diagonal selection for road convert (#13983)
- Fix: Validate raw strings from game-scripts, and strip invalid and control characters (#13976)
- Fix: Capitalise "Disabled" for the "maximum non-sticky open windows" setting (#13975)

> - Fix: Fixed accidental changes to autorail/road AI behavior (#13968)

Reviewer 2: 13968: was broken after beta1.

Reviewer 2: ```suggestion
```

- Fix: Frame widget with label had incorrect spacing (#13967)
- Fix: StringFilter included quotes in the search and failed (#13965)
- Fix #13955: Make graphs respect RTL (#13957)

> - Fix: Changing ownership of a tunnel/bridge did not check for road transport type (#13963)

Reviewer 2: 13963: was broken after beta1

Reviewer 2: ```suggestion
```

- Fix: Numbers were left-aligned for RTL languages in several windows (#13959)
- Fix: MayHaveRoad claimed rail station tiles had road, so the custom stationspec index would be read as roadtype (#13949)
- Fix: [Script] Prevent cloning of API instances (#13947)
- Fix: Reference to the correct section of the README, if a graphics or a sound set is incomplete (#13946)

> - Fix: NewGRF details view does not show the "No information available", if the NewGRF sets an empty description (#13939)

Reviewer 2: 13939: was broken after beta1.

Reviewer 2: ```suggestion
```

- Fix: Draw the bevel around the music track name as inset (#13935)
- Fix #13923: Padding in music GUI was asymmetric, so it looked different for LTR and RTL languages (#13933)

> - Fix #13925: Ancient NewGRF have empty name and description, show the filename instead (#13932)

Reviewer 2: 13932: was broken after beta1.

Reviewer 2: ```suggestion
```
